### ISSUE TITLE
MCPE-1.0: McRegion worlds throw errors

### ISSUE THREAD

**Issue by legoboy0215** [pmmp/PocketMine-MP#232]
### Issue description

McRegion worlds I use throw this error:
```
2017-01-06 [16:17:35] [Server thread/CRITICAL]: TypeError: "Argument 8 passed to pocketmine\level\format\generic\GenericChunk::__construct() must be of the type array, string given, called in phar:///root/pm-1.0/PocketMine-MP.phar/src/pocketmine/level/format/region/McRegion__64bit.php on line 187" (EXCEPTION) in "/src/pocketmine/level/format/generic/GenericChunk" at line 93
2017-01-06 [16:17:35] [Server thread/ERROR]: Corrupted chunk detected
```

### Steps to reproduce the issue

1. Just join the world

### OS and versions

* PocketMine-MP: PocketMine-MP 1.6.2 d890867
* PHP: 7.0.6
* Server OS: Ubuntu 14.10
* Game version: PE/Win10

### Crashdump, backtrace or other files

```
2017-01-06 [16:17:35] [Server thread/CRITICAL]: TypeError: "Argument 8 passed to pocketmine\level\format\generic\GenericChunk::__construct() must be of the type array, string given, called in phar:///root/pm-1.0/PocketMine-MP.phar/src/pocketmine/level/format/region/McRegion__64bit.php on line 187" (EXCEPTION) in "/src/pocketmine/level/format/generic/GenericChunk" at line 93
2017-01-06 [16:17:35] [Server thread/ERROR]: Corrupted chunk detected
```

**Comment by robske110** [comment]
Please include the world. I can load&generate&join MCRegion worlds just fine.

**Comment by legoboy0215** [comment]
Well, it is my server spawn :P

Seems like the world is corrupted? The eighth argument is the height map, and I recently got crashes which involved height maps.

**Comment by robske110** [comment]
If you don't want to share the world there is little anyone can do. And yes, the map is probably corrupted, try force saving the heightmap as an empty array/recreate the heightmap.

**Comment by dktapps** [comment]
This is a common issue, but I haven't been able to reproduce it locally. Could you:

a) provide a reproducing world save
b) try with a non-optimised phar (the preprocessor often causes issues).

**Comment by dktapps** [comment]
So, it seems that in some older worlds, heightmaps were inconsistently saved sometimes as IntArrays and sometimes as ByteArrays. Question: What the #### did this? This is not an issue with the current PocketMine-MP codebase unless I am much mistaken.

**Comment by dktapps** [comment]
#blameshoghi

This is due to world conversion with official tools. In the format described [here](http://minecraft.gamepedia.com/Alpha_level_format#Chunk_Format), HeightMap is saved as a ByteArray. However, for the last 2 years, PocketMine has been reading and writing _IntArrays_. Before I re-implemented chunks for 1.0, this would just cause your world to silently be corrupted. Now, however, it causes a crash due to incorrect types.

**Comment by dktapps** [comment]
Fixed in the referenced commit.
